src/Engine/Audio/SoundBuffer.hpp

```diff
@@ -4,64 +4,58 @@
 #include <json/json.h>
 #include <cstdint>
 #include "../linking.hpp"
+#include <queue>
+#include "SoundStreamer.hpp"
+#include "../Manager/SoundManager.hpp"
 
 namespace Audio {
     class SoundFile;
     
-    /// Wrapper for OpenAL buffers.
+    /// Wrapper for sound file.
     class SoundBuffer {
         public:
-            /// Create new unloaded sound buffer.
+            friend class SoundStreamer;
+
+            /// Constructor.
             ENGINE_API SoundBuffer();
             
-            /// Create a sound buffer from a sound file.
-            /**
-             * @param soundFile The sound file containing the sound.
-             */
-            ENGINE_API explicit SoundBuffer(SoundFile* soundFile);
-            
             /// Destructor.
             ENGINE_API ~SoundBuffer();
             
-            /// Get AL buffer.
+            /// Get sound data.
             /**
-             * @return The OpenAL buffer ID.
+             * @param [OUT] Number of valid samples fetched from file.
```

Comment:
Incorret doxygen comment. `@param` should specify the name of the parameter (`samples`).

Reply:
Fixed